docs/data/material/components/buttons/buttons.md

```diff
@@ -113,12 +113,45 @@ Use `color` prop to apply theme color palette to component.
 
 {{"demo": "IconButtonColors.js"}}
 
+### Loading
+
+Starting from [`v6.4.0`](https://github.com/mui/material-ui/releases/tag/v6.4.0), use `loading` prop to set icon buttons in a loading state and disable interactions.
+
+{{"demo": "LoadingIconButton.js"}}
+
+### Badge
```

Comment:
Why are we adding this demo in this PR?

Reply:
1. to show how Badge can be used with IconButton (it's quite a common use case)
2. to add the demo to visual regression (there was a report about the misalignment)

Reply:
I meant why in this PR and not in another one.

Reply:
Got you. It was added to ensure that the loader (even though not active) does not cause the layout shift to the icon. I got a report from Olivier about the use case, that's why I added in this PR.